app/models/piece.rb

```diff
@@ -15,8 +15,58 @@ def serializable_hash(options = nil)
     super.merge("type" => type)
   end
 
-  def move_to!(row, col)
-    raise "Out of bounds" if row < 0 || row > 7 || col < 0 || col > 7
-    update(row: row, col: col)
+  def move_to!(to_row, to_col)
+    Rails.logger.debug "Model move_to. from location: #{self.row},#{self.col} to location: #{to_row},#{to_col}"
+
+    if valid_move?(to_row,to_col)
+      Rails.logger.debug "valid_move result TRUE. Piece stays at new location. controller update UPDATES table."
+      # use update_attributes in controller update method instead update(row: to_row, col: to_col)
+      self.row = to_row
+      self.col = to_col
+      return true
```

Comment:
You can insert the `update_attributes(row: to_row, col: to_col)` line right before the return true here.  Even better, you can just make this `return update_attributes(row: to_row, col: to_col)`, because `update_attributes` returns true if the DB update succeeds, so you'd get a false if the DB failed for some reason.